extensions/amp-ad/0.1/amp-ad.css

```diff
@@ -48,9 +48,13 @@ amp-embed iframe {
   padding: 2px 4px;
 }
 
-amp-ad[type=adsense] > iframe {
+amp-ad[data-a4a-upgrade-type] > iframe {
+  width: 100%;
+  height: 100%;
   top: 50% !important;
   left: 50% !important;
   transform: translate(-50%, -50%);
+  min-width: 0;
+  min-height: 0;
```

Comment:
I'll leave the correctness of the CSS to you and @keithwrightbos .  All I ask is that you test manually on both AdSense and Doubleclick slots on both Chrome and Safari in a couple of real-world pages.  (Responsive slots, I guess, so that we're sure that the centering behavior applies.)